### PR TITLE
API cleanup

### LINKED ISSUE #137: Python API adheres to requirements
## Definition of Done
The API requirements are fulfilled by its functionalities, any functionality that doesn't contribute to the requirements is eliminated

### PR DESCRIPTION
**Pull request by bkmartinjr**
Fixes #137 

Changes:
* removed get_experiment
* added module-level help string
* fixed a few docstring typos
* renamed private methods to include underscore prefix

### CODE REVIEW

**Comment by pablo-gar** [comment]
@ebezzi there's some doc string work here related to #140

File: api/python/cell_census/src/cell_census/__init__.py [hunk 1]
```diff
@@ -1,12 +1,24 @@
+"""
+Cell Census convience API
+
+An API to facilitate use of the CZI Science Cell Census. The Cell Census is a versioned container of single-cell data hosted at [CELLxGENE Discover](https://cellxgene.cziscience.com/).
+
+The API is built on the `tiledbsoma` SOMA API, and provides a number of helper functions including:
+
+    * Open a named version of the Cell Census, for use with the SOMA API
+    * Get a list of available Cell Census verisons, and for each version, a description
```
Comment (ebezzi) [inline api/python/cell_census/src/cell_census/__init__.py:9]:
```suggestion
    * Get a list of available Cell Census versions, and for each version, a description
```

**Comment by bkmartinjr** [comment]
@ebezzi - at Pablo's request, I stopped all of my work on the docstrings

File: api/python/cell_census/src/cell_census/__init__.py (hunk 1 above)
Comment (bkmartinjr) [reply api/python/cell_census/src/cell_census/__init__.py]: fixed.  ty for the proofing!

**Comment by ebezzi** [comment]
> @ebezzi - at Pablo's request, I stopped all of my work on the docstrings

I am happy to pick up the work myself but since my knowledge about the API is still quite limited, if there are any specific changes you have in mind, I would appreciate pointers. My plan for now would be to go through all the methods/classes and see if there are any docstrings missing.

**Comment by bkmartinjr** [comment]
@ebezzi - what I know:
1. all public functions have docstrings.  It is good for you to validate this, and even better if you proof-read them as a "new user" for clarity and sufficient information/example to be useful for other new users.
2. if you find any issues/gaps and you need help with content, just let me know and I'll generate suggestions and/or new content - whatever works best for you!
Thanks for doing this!

**Review by Bento007 — APPROVED** [review]

**Comment by Bento007** [comment]
Was this actually ready for review? It looks like a complete chunk. Was there more you were planning on adding?

**Comment by bkmartinjr** [comment]
> Was this actually ready for review? It looks like a complete chunk. Was there more you were planning on adding?

Yep, it was ready. 

I use the standard GH convention of leaving PRs as a Draft PR until they are ready for review. When ready, they are marked "Ready for Review"

File: api/python/cell_census/src/cell_census/get_anndata.py [hunk 2]
```diff
@@ -25,9 +25,7 @@ def get_anndata(
 ) -> anndata.AnnData:
     """
     Convience wrapper around soma.Experiment query, to build and execute a query,
-    and return it as an AnnData object.
-
-    [lifecycle: experimental]
```
Comment (atolopko-czi) [inline api/python/cell_census/src/cell_census/get_anndata.py:30]: rm of lifecycle intentional?

**Review by atolopko-czi — APPROVED** [review]
LGTM

File: api/python/cell_census/src/cell_census/get_anndata.py (hunk 2 above)
Comment (bkmartinjr) [reply api/python/cell_census/src/cell_census/get_anndata.py:30]: it was not removed.  Just merged with previous line.

**Review by pablo-gar — APPROVED** [review]

**Review by ebezzi — APPROVED** [review]

**Pull request merged**